### PR TITLE
Added Top_p https://github.com/uripeled2/llm-client-sdk/issues/28

### LINKED ISSUE #28: Add p-top to common models parameter
Add p-top (here is OpenAI docs on [p-top](https://platform.openai.com/docs/api-reference/chat/create#chat/create-top_p)) as a param in BaseLLMAPIClient.text_completion.
You can take a look at this great [PR](https://github.com/uripeled2/llm-client-sdk/commit/11c56055c4d67bae9c9a6b8b49faad8a6fbe9eed) done by @EyalPaz-700 and see how he added max_tokens and temperature into BaseLLMAPIClient.text_completion params

### PR DESCRIPTION
**Pull request by EyalPazz**

### CODE REVIEW

**Comment by uripeled2** [comment]
You didn't update BaseLLMAPIClient.text_completion function

File: llm_client/llm_api_client/anthropic_client.py [hunk 1]
```diff
@@ -27,13 +27,14 @@ def __init__(self, config: LLMAPIClientConfig):
         self._headers[AUTH_HEADER] = self._api_key
 
     async def text_completion(self, prompt: str, model: Optional[str] = None, max_tokens: Optional[int] = None,
-                              temperature: float = 1,
+                              temperature: float = 1, top_p: float = -1,
```
[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/anthropic_client.py:30]: https://autocode.com/anthropic/api/playground/dev/complete/

[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/anthropic_client.py:30]: idk if it's None tbh

File: llm_client/llm_api_client/ai21_client.py [hunk 2]
```diff
@@ -22,9 +22,10 @@ def __init__(self, config: LLMAPIClientConfig):
         self._headers[AUTH_HEADER] = BEARER_TOKEN + self._api_key
 
     async def text_completion(self, prompt: str, model: Optional[str] = None, max_tokens: int = 16,
-                              temperature: float = 0.7, **kwargs) -> list[str]:
+                              temperature: float = 0.7, top_p: Optional[float] = None ,**kwargs) -> list[str]:
```
[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/ai21_client.py:25]: it's weird that i missed that but cool i will push soon

File: llm_client/llm_api_client/aleph_alpha_client.py [hunk 3]
```diff
@@ -27,12 +27,13 @@ def __init__(self, config: LLMAPIClientConfig):
         self._headers[AUTH_HEADER] = BEARER_TOKEN + self._api_key
 
     async def text_completion(self, prompt: str, model: Optional[str] = None, max_tokens: Optional[int] = None,
-                              temperature: float = 0, **kwargs) -> \
+                              temperature: float = 0,top_p: Optional[float] = None, **kwargs) -> \
```
[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/aleph_alpha_client.py:30]: it's weird that i missed that but cool i will push soon

File: tests/resources/openai/chat_completion.json [hunk 4]
```diff
@@ -13,6 +13,7 @@
   "usage": {
     "prompt_tokens": 9,
     "completion_tokens": 12,
-    "total_tokens": 21
+    "total_tokens": 21,
+        "top_p" : 1
```
[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply tests/resources/openai/chat_completion.json:17]: shouldn't it be here?

[the remark this answers is not in the document]
Comment (uripeled2) [reply tests/resources/openai/chat_completion.json:17]: this is the response not the request

**Comment by uripeled2** [comment]
@CodiumAI-Agent

File: tests/llm_api_client/anthropic_client/test_anthropic_client.py
Comment (uripeled2) [inline tests/llm_api_client/anthropic_client/test_anthropic_client.py:1]: add top_p into test_text_completion__with_kwargs

File: llm_client/llm_api_client/google_client.py [hunk 5]
```diff
@@ -33,10 +33,11 @@ def __init__(self, config: LLMAPIClientConfig):
         self._params = {AUTH_PARAM: self._api_key}
 
     async def text_completion(self, prompt: str, model: Optional[str] = None, max_tokens: Optional[int] = 64,
-                              temperature: Optional[float] = None, **kwargs) -> list[str]:
+                              temperature: Optional[float] = None,top_p: float = 0.95, **kwargs) -> list[str]:
```
[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/google_client.py:36]: https://cloud.google.com/vertex-ai/docs/generative-ai/start/quickstarts/api-quickstart

[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/google_client.py:36]: Note: The default value varies by model, see the Model.top_p attribute of the Model returned the getModel function.

[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/google_client.py:36]: https://cloud.google.com/vertex-ai/docs/generative-ai/start/quickstarts/api-quickstart

[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/google_client.py:36]: maybe we'll do it like the anthropic one, like that :
        if top_p:
            kwargs["top_p"] = top_p

[the remark this answers is not in the document]
Comment (uripeled2) [reply llm_client/llm_api_client/google_client.py:36]: Agree, let's do it like Anthropic

File: tests/llm_api_client/anthropic_client/test_anthropic_client.py
Comment (EyalPaz-700) [reply tests/llm_api_client/anthropic_client/test_anthropic_client.py:1]: done

File: llm_client/llm_api_client/google_client.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/google_client.py]: done

File: llm_client/llm_api_client/google_client.py [hunk 6]
```diff
@@ -33,11 +33,12 @@ def __init__(self, config: LLMAPIClientConfig):
         self._params = {AUTH_PARAM: self._api_key}
 
     async def text_completion(self, prompt: str, model: Optional[str] = None, max_tokens: Optional[int] = 64,
-                              temperature: Optional[float] = None,top_p: float = 0.95, **kwargs) -> list[str]:
+                              temperature: Optional[float] = None,top_p: Optional[float] = None, **kwargs) -> list[str]:
         model = model or self._default_model
         kwargs[PROMPT_KEY] = {TEXT_KEY: prompt}
         kwargs[MAX_TOKENS_KEY] = kwargs.pop(MAX_TOKENS_KEY, max_tokens)
-        kwargs["topP"] = kwargs.pop("topP", top_p)
+        if top_p or ("topP" in kwargs):
```
[the remark this answers is not in the document]
Comment (EyalPaz-700) [reply llm_client/llm_api_client/google_client.py:40]: yeah my bad :(

**Pull request merged**